lib/experiments-page.php

```diff
@@ -0,0 +1,119 @@
+ */
+function the_gutenberg_experiments( $page = 'gutenberg_page_gutenberg-experiments' ) {
+	?>
+	<div
+		id="experiments-editor"
+		class="wrap"
+	>
+	<h1><?php echo __( 'Experiment settings', 'gutenberg' ); ?></h1>
+	<?php settings_errors(); ?>
+	<form method="post" action="options.php">
+		<?php settings_fields( 'gutenberg-experiments' ); ?>
+		<?php do_settings_sections( 'gutenberg-experiments' ); ?>           
+		<?php submit_button(); ?>
+	</form>
+	</div>
+	<?php
+}
+
+/**
+ * Set up the experiments settings.
+ *
+ * @since 5.2.3
+ */
+function gutenberg_initialize_experiments_settings() {
+	add_settings_section(
+		'gutenberg_experiments_section',
+		// The empty string ensures the render function won't output a h2.
+		'',
+		'gutenberg_display_experiment_section',
+		'gutenberg-experiments'
+	);
+	add_settings_field(
+		'gutenberg-widget-experiments',
+		__( 'Widgets', 'gutenberg' ),
+		'gutenberg_display_experiment_field',
+		'gutenberg-experiments',
+		'gutenberg_experiments_section',
+		array(
+			'label' => __( 'Enable Widgets Screen and Legacy Widget Block', 'gutenberg' ),
+			'id'    => 'gutenberg-widget-experiments',
+		)
+	);
+	add_settings_field(
+		'gutenberg-menu-block',
+		__( 'Menu Block', 'gutenberg' ),
+		'gutenberg_display_experiment_field',
+		'gutenberg-experiments',
+		'gutenberg_experiments_section',
+		array(
+			'label' => __( 'Enable Navigation Menu Block', 'gutenberg' ),
+			'id'    => 'gutenberg-menu-block',
+		)
+	);
+	register_setting(
+		'gutenberg-experiments',
+		'gutenberg-experiments'
+	);
+}
+
+add_action( 'admin_init', 'gutenberg_initialize_experiments_settings' );
+
+/**
+ * Display a checkbox field for a Gutenberg experiment.
+ *
+ * @since 5.2.3
+ *
+ * @param array $args ( $label, $id ).
+ */
+function gutenberg_display_experiment_field( $args ) {
+	$options = get_option( 'gutenberg-experiments' );
+	$value   = isset( $options[ $args['id'] ] ) ? 1 : 0;
+	?>
+		<label for="<?php echo $args['id']; ?>">
+			<input type="checkbox" name="<?php echo 'gutenberg-experiments[' . $args['id'] . ']'; ?>" id="<?php echo $args['id']; ?>" value="1" <?php checked( 1, $value ); ?> />
+			<?php echo $args['label']; ?>
+		</label>
+	<?php
+}
+
+/**
+ * Display the experiments section.
+ *
+ * @since 5.2.3
+ */
+function gutenberg_display_experiment_section() {
+	?>
+	<p><?php echo __( 'Gutenberg has some experimental features you can turn on. Simply select each you would like to use. These features are likely to change so it is inadvisable to use them in production.', 'gutenberg' ); ?></p>
+
+	<?php
+}
+
+/**
+ * Extends default editor settings with experiments settings.
+ *
+ * @param array $settings Default editor settings.
+ *
+ * @return array Filtered editor settings.
+ */
+function gutenberg_experiments_editor_settings( $settings ) {
+	$experiments_exist = get_option( 'gutenberg-experiments' );
+	return array(
```

Comment:
I was wondering why so many e2e tests are failing in this branch. I think this is the cause. The $settings being passed in need to be merged with the array of new settings. You should be able to use `array_merge`.